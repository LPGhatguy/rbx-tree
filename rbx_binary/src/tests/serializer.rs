use rbx_dom_test::InstanceBuilder;

use crate::{debug_deserializer::DecodedModel, encode};

#[test]
fn folder_from_studio() {
    static FOLDER: &[u8] = include_bytes!("../../test-files/just-folder.rbxm");

    let decoded = DecodedModel::from_reader(FOLDER);
    insta::assert_yaml_snapshot!(decoded);
}

#[test]
fn just_folder() {
    let tree = InstanceBuilder::new("Folder").build();
    let mut buffer = Vec::new();

    encode(&tree, &[tree.get_root_id()], &mut buffer).expect("failed to encode model");

    let decoded = DecodedModel::from_reader(buffer.as_slice());
    insta::assert_yaml_snapshot!(decoded);
}
