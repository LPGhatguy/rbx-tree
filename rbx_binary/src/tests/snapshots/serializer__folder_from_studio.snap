---
source: rbx_binary/src/tests/serializer.rs
expression: decoded
---
num_types: 1
num_instances: 1
chunks:
  - Meta:
      num_entries: 1
      remaining: EgAAAEV4cGxpY2l0QXV0b0pvaW50cwQAAAB0cnVl
  - Inst:
      remaining: AAAAAAYAAABGb2xkZXIAAQAAAAAAAAA=
  - Prop:
      remaining: AAAAABMAAABBdHRyaWJ1dGVzU2VyaWFsaXplAQAAAAA=
  - Prop:
      remaining: AAAAAAQAAABOYW1lAQYAAABGb2xkZXI=
  - Prop:
      remaining: AAAAAAQAAABUYWdzAQAAAAA=
  - Prnt:
      remaining: AAEAAAAAAAAAAAAAAQ==
  - End
